### PR TITLE
Improve Helm Documentation and Fix Tests

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description
- Update Helm install README to clarify behavior
- Fix Helm unit tests
- Fix github action permissions
- Fix upgrade test NPE

# Context

## Helm Unit Tests
There was a [previous bug](https://github.com/solo-io/gloo/issues/7063) that caused Helm tests to not run locally.

We use `make print-VERSION` to identify the VERSION variable that will be executed during the test. The utility in our test however worked for forks, or in CI, but not locally.

To strategize the fix, I first ran some our CI pipeline against a fork, and gathered some [useful logs](https://github.com/solo-io/gloo/pull/7538#issuecomment-1345665694).

Then I ran the tests locally and was able to identify the 2 formats that `make print-VERSION` can return.

We now handle both.

### Testing
Run the tests locally `TEST_PKG=install/test make test`

## Github Action Permissions
Define permission based environment variables at the workflow level, so that we only define them in one place. Also, use the same secret for the github token.

## Upgrade Test NPE
By fixing the permission issue (above), we uncovered an NPE in the upgrade tests (https://github.com/solo-io/gloo/actions/runs/3671275039/jobs/6206370988)

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/kube2e/README.md [hunk 1]
```diff
@@ -19,45 +19,34 @@ If a test fails, you can retry it from a [browser window](https://docs.github.co
...
+| CLUSTER_NODE_VERSION | v1.22.4  | The version of the [Node Docker image](https://hub.docker.com/r/kindest/node/) to use for booting the cluster       |
+| VERSION              | 1.0.0-ci | The version used to tag Gloo images that are deployed to the cluster                                                |
+| KUBE2E_TESTS         | gateway  | Name of the test suite to be run. Options: `'gateway', 'gloo', 'ingress', 'helm', 'gloomtls', 'glooctl', 'upgrade'` |
 
 Example:
 ```bash
 CLUSTER_NAME=solo-test-cluster CLUSTER_NODE_VERSION=v1.22.4 VERSION=v1.0.0-solo-test ci/deploy-to-kind-cluster.sh
 ```
 
-#### (Option B) - Manually Run Make Targets
```
Comment (sam-heilbron) [inline test/kube2e/README.md:44]: The setup script has become more complex that running the steps manually is harder to keep up to date. I think having a single recommended path is easier to reason through. If somone wants to run the steps manually, they can inspect the script to understand what is being done

File: test/kube2e/util.go [hunk 2]
```diff
@@ -290,12 +287,6 @@ func ToFile(content string) string {
 	return f.Name()
 }
 
-// PatchResource mutates an existing resource, retrying if a resourceVersionError is encountered
-// Deprecated: Prefer the helpers.PatchResource, which is not a Kubernetes specific package
-func PatchResource(ctx context.Context, resourceRef *core.ResourceRef, mutator func(resource resources.Resource), client clients.ResourceClient) error {
```
Comment (gunnar-solo) [inline test/kube2e/util.go:295]: cracking me up as these calls gradually go extinct across our multiple suites

File: install/README.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
+# Installing Gloo Edge
+
+## Installing on a Local Filesystem
```
Comment (gunnar-solo) [inline install/README.md:3]: suggestion:
* [Installing on a Local Filesystem](https://docs.solo.io/gloo-edge/latest/installation/gateway/development/docker-compose-file/)
* [Installing on a Consul](https://docs.solo.io/gloo-edge/latest/installation/gateway/development/docker-compose-consul/)

File: install/README.md [hunk 4]
```diff
@@ -1,3 +1,11 @@
...
+Refer to our [external documentation](https://docs.solo.io/gloo-edge/latest/installation/gateway/development/docker-compose-file/)
+
+## Installing on a Consul
+Refer to our [external documentation](https://docs.solo.io/gloo-edge/latest/installation/gateway/development/docker-compose-consul/)
+
 # Installing on Kubernetes
 
 Note: if running on GKE, you need to configure permission to create rbac:
```
Comment (gunnar-solo) [inline install/README.md:11]: if making changes to this file, I recommend throwing a

⚠️ ${content} ⚠️

around this warning

File: install/helm/README.md [hunk 5]
```diff
@@ -1,7 +1,10 @@
 # Gloo Edge Helm chart
 This directory contains the resources used to generate the Gloo Edge Helm chart archive.
 
-## generate.go
+> All make targets are currently defined in the [Makefile](https://github.com/solo-io/gloo/blob/master/Makefile) and should be executed from the root of the repository.
```
Comment (gunnar-solo) [inline install/helm/README.md:4]: content suggestion:

📝 [make targets](https://opensource.com/article/18/8/what-how-makefile]) are currently defined in the [Makefile](https://github.com/solo-io/gloo/blob/master/Makefile) and should be executed from the root of the repository 📝

File: install/helm/README.md [hunk 6]
```diff
@@ -11,9 +14,51 @@ to generate the following files:
...
-## /crds
+### /crds
 This directory contains the Gloo Edge `CustomResourceDefinitions`. This is the 
 [required location](https://helm.sh/docs/topics/charts/#custom-resource-definitions-crds) for CRDs in Helm 3 charts.
 
-## /templates
+### /templates
 This directory contains the Helm templates used to generate the Gloo Edge manifests.
+
+## Build
```
Comment (gunnar-solo) [inline install/helm/README.md:24]: When reading/writing docs, I try to stay away from so many header sections, as I find that, as they increase in number, the visual clutter weakens a given doc.  Full suggestion for the remainder of the file:


----------------------------


## Helm-centric commands
Relevant commands to helm, meant to be run from the **root of this repository**

```bash
# VERSION is an optional environment variable.  If not specified, a default will be computed
VERSION=$VERSION make generate-helm-files    # generate `Chart.yaml` and `values.yaml` files
VERSION=$VERSION make package-chart          # package a helm chart to `_output/charts` directory (used for releasing)
VERSION=$VERSION make build-test-chart       # package a helm chart to `_test` directory (used for testing)

helm install gloo gloo/gloo                  # install Gloo Edge using Helm
TEST_PKG=install/test make test              # run all tests in this project

```

Further reading:
- What is [helm](https://helm.sh/docs/helm/helm_install/)?
- How do I [install on kubernetes with helm](https://docs.solo.io/gloo-edge/latest/installation/gateway/kubernetes/#installing-on-kubernetes-with-helm)?
- What is a [packaged Chart archive](https://helm.sh/docs/helm/helm_package/)?
- Where are our [gloo charts published](https://storage.googleapis.com/solo-public-helm) to?

File: test/kube2e/README.md (hunk 1 above)
Comment (gunnar-solo) [reply test/kube2e/README.md:44]: good idea!  No opposition, here

**Review by gunnar-solo — APPROVED** [review]

**Review by elcasteel — APPROVED** [review]

**Pull request merged**